lightning/src/ln/channelmanager.rs

```diff
@@ -2659,6 +2659,16 @@ where
 		amt_msat: u64, cltv_expiry: u32, phantom_shared_secret: Option<[u8; 32]>, allow_underpay: bool,
 		counterparty_skimmed_fee_msat: Option<u64>,
 	) -> Result<PendingHTLCInfo, InboundOnionErr> {
+		let (payment_data, keysend_preimage, payment_metadata) = match hop_data.format {
+			msgs::OnionHopDataFormat::FinalNode { payment_data, keysend_preimage, payment_metadata } =>
+				(payment_data, keysend_preimage, payment_metadata),
+			_ =>
+				return Err(InboundOnionErr {
+					err_code: 0x4000|22,
+					err_data: Vec::new(),
+					msg: "Got non final data with an HMAC of 0",
+				}),
```

Comment:
Does it matter that this check happens before other checks now?

Reply:
What I meant was, could we return a different error now if the order of the error checks have changed? Or does something preclude that? (i.e., the conditions leading to them do not overlap)

Reply:
Yeah, we might return a different error now if multiple errors apply. I don't think it matters, though. The one error that's in a different order now is the `Got non final data with HMAC of 0`, which should be really rare/indicate a very buggy sender.